jsonlinter.py

```diff
@@ -0,0 +1,98 @@
+# python3 lintjson.py
+# William Rehwinkel / obiesource, 2022
+
+import sys
+import os
+import json
+import traceback
+
+def getJsonFiles():
+    # list all files recursively borrowed from https://pythonexamples.org/python-get-list-of-all-files-in-directory-and-sub-directories/
+    filelist = list()
+    path = "./"
+    
+    for root, dirs, files in os.walk(path):
+	    for file in files:
+            #append the file name to the list
+		    filelist.append(os.path.join(root,file))
+    
+    #print all the file names
+    jsonFiles = list()
+    
+    for name in filelist:
+        if name.lower().endswith(".json"):
+            jsonFiles.append(name)
+    return jsonFiles
+
+MEMBERS_REQUIRED = {
+        "name": "",
+        "pronouns": "",
+        "class": "",
+        "socials": list(),
+        "bio": "",
+        "projects": list()
+        }
+PROJECTS_REQUIRED = {
+        "name": "",
+        "description": "",
+        "website": ""
+        }
+
+def lintJson(filename):
+    print("Checking file " + filename)
+    with open(filename) as file:
+        try: js = json.load(file)
+        except Exception as e:
+            print("Error while parsing " + filename)
+            traceback.print_tb(e)
+            return True
+
+        failed = False
+        if filename.startswith("./members/"):
+            # check for required keys
+            for k in MEMBERS_REQUIRED:
+                t = MEMBERS_REQUIRED[k]
+                if k not in js:
+                    print("Key " + k + " required but not found in members JSON file.")
+                    failed = True
+                elif type(js[k]) != type(t):
+                    print("Key " + k + " has incorrect type " + str(type(js[k])) + ", needs " + str(type(t)))
```

Comment:
```suggestion
                elif type(js[k]) != t:
                    print("Key " + k + " has incorrect type " + str(type(js[k])) + ", needs " + str(t))
```
Commit this if you (@FiskFan1999) accept the change [here](https://github.com/ObieSource/obiesource.github.io/pull/9/files#r842743262)